lib/src/components/NxTag/NxTag.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { FunctionComponent } from 'react';
+import classnames from 'classnames';
+import { faPlusCircle } from '@fortawesome/free-solid-svg-icons';
+
+import './NxTag.scss';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import NxOverflowTooltip from '../NxTooltip/NxOverflowTooltip';
+import { Props, propTypes } from './types';
+export { Props } from './types';
+
+const NxTag: FunctionComponent<Props> =
+    function NxTag(props) {
+      const { children, className, tagColor, ...attrs } = props,
+          tagClasses = classnames('nx-tag', className, {
+            [`nx-tag--${tagColor}`]: tagColor,
+            ['nx-tag--default']: !tagColor
+          });
+
+      return (
+        <div className={tagClasses} {...attrs}>
+          <NxOverflowTooltip>
+            <div className="nx-tag__text">{children}</div>
+          </NxOverflowTooltip>
+        </div>
+      );
+    };
+
+NxTag.propTypes = propTypes;
+export default NxTag;
+
+export const NxSelectableTag: FunctionComponent<Props> =
+    function NxSelectableTag(props) {
+      const { children, className, tagColor, tagSelected, onTagSelect, ...attrs } = props,
+          isSelected = tagSelected,
+          tagClasses = classnames('nx-tag nx-tag--selectable', className, {
+            'nx-tag--selected': isSelected,
+            'nx-tag--unselected': !isSelected,
+            [`nx-tag--${tagColor}`]: tagColor,
+            ['nx-tag--default']: !tagColor
+          });
+
+      return (
+        <div tabIndex={0}
+             role="switch"
+             aria-checked={tagSelected}
+             className={tagClasses}
+             onClick={onTagSelect || undefined}
+             {...attrs}>
+          <NxOverflowTooltip>
+            <div className="nx-tag__text">{children}</div>
+          </NxOverflowTooltip>
+          <div className="nx-tag__action">
+            <NxFontAwesomeIcon icon={faPlusCircle} />
+          </div>
+        </div>
+      );
+    };
```

Comment:
I feel like there's potential for code sharing between the two tag implementations.  Particularly, that `NxSelectableTag` could be built on top of `NxTag`, particularly if `nx-tag__text` was made optional which could happen if my advice in `https://github.com/sonatype/sonatype-react-shared-components/pull/290/files#r562781459` is followed.  ie `NxTag` would just include its children directly and only `NxSelectableTag` would wrap the children in `nx-tag__text`.

Reply:
> I feel like there's potential for code sharing between the two tag implementations. Particularly, that `NxSelectableTag` could be built on top of `NxTag`, particularly if `nx-tag__text` was made optional which could happen if my advice in `https://github.com/sonatype/sonatype-react-shared-components/pull/290/files#r562781459` is followed. ie `NxTag` would just include its children directly and only `NxSelectableTag` would wrap the children in `nx-tag__text`.

I'm happy with how it is now.

Reply:
> I'm happy with how it is now.

Why? It's a pretty small change from where you have things now: https://github.com/sonatype/sonatype-react-shared-components/pull/294